### PR TITLE
Return more descriptive error when JWK matching `kid` not found in JWKS

### PR DESCRIPTION
**Pull request by strideynet**
Closes https://github.com/go-jose/go-jose/issues/127

Returns a more descriptive error as the current error is fairly cryptic when the KID header within a JWT does not match the KID of any JWK in a JWKS.

### CODE REVIEW

File: jwk.go [hunk 1]
```diff
@@ -779,7 +779,13 @@ func (key rawJSONWebKey) symmetricKey() ([]byte, error) {
 	return key.K.bytes(), nil
 }
 
-func tryJWKS(key interface{}, headers ...Header) interface{} {
+var (
+	// ErrJWKSKidNotFound is returned when a JWKS does not contain a JWK with a
+	// key ID which matches one in the provided tokens headers.
+	ErrJWKSKidNotFound = errors.New("go-jose/go-jose: JWK with matching kid not found in JWKS")
```
Comment (jsha) [inline jwk.go:785]: Nit:

```suggestion
	ErrJWKSKidNotFound = errors.New("go-jose/go-jose: JWK with matching kid not found in JWK Set")
```

[The RFC uses "JWS Set"](https://datatracker.ietf.org/doc/html/rfc7517#section-5) and doesn't seem to abbreviate it to JWKS anywhere.

File: jwt/jwt_test.go [hunk 2]
```diff
@@ -111,6 +112,27 @@ func TestDecodeTokenWithJWKS(t *testing.T) {
 	}
 }
 
+// TestDecodeTokenWithMismatchedJWKSKID tests a case where the JWT has a KID
+// header which does not match any of the keys in the JWKS.
+func TestDecodeTokenWithMismatchedJWKSKID(t *testing.T) {
```
Comment (jsha) [inline jwt/jwt_test.go:117]: Could you also add a test case where KeyID is empty?

**Review by jsha — CHANGES_REQUESTED** [review]
Looks great. Thanks for the clear explanation and the accompanying PR. A couple small notes below.

**Review by jsha — APPROVED** [review]
(I went ahead and pushed those small changes)

File: jwk.go (hunk 1 above)
Comment (strideynet) [reply jwk.go]: Awesome - I'm not deeply familiar with the spec but this makes sense.

**Review by mcpherrinm — APPROVED** [review]

**Review by aarongable — APPROVED** [review]
